presto-main/src/main/java/com/facebook/presto/operator/WindowOperator.java

```diff
@@ -193,6 +193,7 @@ public OperatorFactory duplicate()
     private final PageBuilder pageBuilder;
 
     private final DriverWindowInfoBuilder windowInfo;
+    private DriverWindowInfo driverWindowInfo;
```

Comment:
This already happens.
Expensive part (building `DriverWindowInfo`) happens on operator close only.
See b570455.
Until the operator closes, it produces empty `WindowInfo` which is wrapper around `List<DriverWindowInfo>`.
If you don't like it (i.e. claim it's too expensive to create an empty object with `new`, which I doubt is a problem in Java), I can return `null` there or static empty `WindowInfo` instead.